common/src/file_slice.rs

```diff
@@ -32,6 +34,42 @@ pub trait FileHandle: 'static + Send + Sync + HasLen + fmt::Debug {
     }
 }
 
+#[derive(Debug)]
+pub struct WrapFile(pub File);
+
+#[async_trait]
+impl FileHandle for WrapFile {
+    fn read_bytes(&self, range: Range<usize>) -> io::Result<OwnedBytes> {
+        let mut file = self.0.try_clone()?; // Clone the file to read from it separately
```

Comment:
This is probably a premature optimization for elsewhen, but at least on Unix, positioned I/O using e.g. [`read_exact_at`](https://doc.rust-lang.org/stable/std/os/unix/fs/trait.FileExt.html#method.read_exact_at) might avoid this duplication of the file descriptor.

Reply:
For now, I'd just stick with one implementation. In the merge cases this is not really relevant, where this is planned to be used